docs/source/howtos/programmatic_query.rst

```diff
@@ -0,0 +1,324 @@
+searched for is an alias for a particular version of an instrument, that version
+is returned instead of the default version:
+
+>>> available_versions, default_version = Instrument.available_versions('TFXA')
+>>> available_versions
+['TFXA', 'TOSCA1', 'TOSCA']
+>>> default_version
+'TFXA'
+
+.. _how-to-model:
+
+How To Find the Models Available for an Instrument
+--------------------------------------------------
+
+Given an :ref:`instrument<how-to-instrument>` and its
+:ref:`version<how-to-version>`, it is possible to query the list of available
+:term:`models<model>` by using the
+:py:meth:`~resolution_functions.instrument.Instrument.available_models`
+property:
+
+>>> from resolution_functions import Instrument
+>>> tosca = Instrument.from_default('TOSCA', 'TOSCA')
+>>> tosca.available_models
+['AbINS', 'book', 'vision']
+
+The default model can similarly be accessed via an attribute:
+
+>>> tosca.default_model
+'AbINS'
+
+Advanced
+^^^^^^^^
+
+The above property returns only the recommended (usually latest) version of each
+model. These "models" are actually aliases that each point to a versioned model
+that actually holds the data. These versioned models can be listed by using
+:py:meth:`~resolution_functions.instrument.Instrument.available_unique_models`:
+
+>>> tosca.available_unique_models
+['AbINS_v1', 'book_v1', 'vision_v1']
+
+Then, to round out the options, *all* models can be listed via
+:py:meth:`~resolution_functions.instrument.Instrument.all_available_models`:
+
+>>> tosca.all_available_models
+['AbINS', 'AbINS_v1', 'book', 'book_v1', 'vision', 'vision_v1']
+
+.. important::
+
+    Older versions of models often contain bugs and/or inaccuracies - use at own
+    risk.
+
+
+.. _how-to-config:
+
+How To Find the Configurations that Must be Chosen for a Model
+--------------------------------------------------------------
+
+To do this, the :term:`model` must already be known, either from prior knowledge
+or by browsing the :ref:`available models<how-to-model>`; without this decision
+made, the only option is use :ref:`howto-query-models-configs` which returns a
+table. Otherwise, the :term:`configurations<configuration>` can be retrieved
+using the
+:py:meth:`~resolution_functions.instrument.Instrument.possible_configurations_for_model`
+method:
+
+>>> from resolution_functions import Instrument
+>>> tosca = Instrument.from_default('TOSCA', 'TOSCA')
+>>> tosca.possible_configurations_for_model('AbINS')
+[]
+>>> tosca.possible_configurations_for_model('book')
+['detector_bank']
+
+
+.. _howto-query-options:
+
+How To Find the Options Available for a Configuration
+-----------------------------------------------------
+
+Similarly, to obtain the :term:`options<option>` in a Python format, the
+:term:`model` must once again be known
+(:ref:`otherwise<howto-query-models-configs-options>` only a formatted table can
+be returned), though in this case there are multiple options:
```

Comment:
```suggestion
To obtain the :term:`options<option>` as Python data, the
:term:`model` must already be known.
(Humans can read the formatted table from :ref:`otherwise<howto-query-models-configs-options>`.)
```